### PR TITLE
Added sortingTriangles function, unit tests, and trianglesSort handler

### PR DESCRIPTION
**Pull request by Metalscreame**
changed .gitgnore
changed main function and router

### CODE REVIEW

File: src/models/trianglesSort/triangles.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+import (
+	"math"
+	"sort"
+	"errors"
+	"regexp"
+)
+
+const (
+	errorSigned    = "Numbers cant be signed, or float or 0 or string"
```
Comment (oreuta) [inline src/models/trianglesSort/triangles.go]: 1) `"can't"` is OK
2) `errSigned` or better `errBadFormat`

File: src/models/trianglesSort/triangles.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+func validateTriangles(t []Triangle) (err error) {
+	if len(t) == 0 || len(t) == 1 {
+		return errors.New(errorInputSize)
+	}
+	for i := range t {
+		err = t[i].validateSingleTriangle()
+		if err != nil {
+			return err
```
Comment (oreuta) [inline src/models/trianglesSort/triangles.go]: just `return`

File: src/models/trianglesSort/triangles.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+type Triangle struct {
+	Vertices string  `json:"vertices"`
+	A        float64 `json:"a"`
+	B        float64 `json:"b"`
+	C        float64 `json:"c"`
+	Sqrt     float64 `json:"square,-"`
+}
+
+// TrianglesSquareSort is a function that sorts triangles slice by single triangle area. Returns sorted slice and error
+func TrianglesSquareSort(trianglesToSortSlice []Triangle) ([]Triangle, error) {
```
Comment (oreuta) [inline src/models/trianglesSort/triangles.go]: named return rules here

File: src/models/trianglesSort/triangles.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+		err = t[i].calculateSquare()
+		if err != nil {
+			return err
+		}
+	}
+	return
+}
+
+func (t *Triangle) validateSingleTriangle() (err error) {
+	var regex = regexp.MustCompile(`^[A-C]{3}$`) //своя реализация в анмаршалинге
```
Comment (oreuta) [inline src/models/trianglesSort/triangles.go]: 1) pls, place it outside the func
2) English

File: src/models/trianglesSort/triangles.go [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+	}
+	if !t.validateNamesFitNumbers() {
+		return errors.New("Names dont fit actual values")
+	}
+	return
+}
+
+func (t *Triangle) validateNamesFitNumbers() bool {
+	r := []rune(t.Vertices)
+	if string(r[2]) == "A" && t.A > t.B && t.A > t.C {
```
Comment (oreuta) [inline src/models/trianglesSort/triangles.go]: code duplication: this `if` could be placed into a separate function and called three times for A, B & C

File: src/models/trianglesSort/triangles.go (hunk 3 above)
Comment (Metalscreame) [reply src/models/trianglesSort/triangles.go]: i'm returning the same slice as i have for my input, so i found it not necessary (if its named then then i need to reassign old to new which). So should i?

**Pull request closed without merging**